src/shared/Microsoft.Git.CredentialManager/Commands/ProviderCommand.cs

```diff
@@ -0,0 +1,24 @@
+using System.CommandLine;
+
+namespace Microsoft.Git.CredentialManager.Commands
+{
+    public interface ICommandProvider
+    {
+        /// <summary>
+        /// Configure a custom provider command.
+        /// </summary>
+        /// <param name="rootCommand">Root provider command.</param>
+        void ConfigureCommand(Command rootCommand);
+    }
+
+    internal class ProviderCommand : Command
```

Comment:
I'm a little confused as to why these are in the same file. They are not using each other in any visible way.

Reply:
That's because I'm an idiot 😜 

I changed the extension point design here a few times, but this doesn't "feel" right. I've moved to a factory style now where an `ICommandProvider` implementing host provider returns a new `ProviderCommand` instance from the `CreateCommand()` method.